aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js

```diff
@@ -129,32 +129,27 @@ define(["dojo/_base/declare",
       },
 
       /**
-       *
+       * Handles requests to manage aspects by making a request to get the currently applied aspects for the 
+       * requested Node. This is always performed to ensure that the data is not stale.
+       * 
+       * @instance
        * @param {object} item The item to perform the action on
        */
       onManageAspects: function alfresco_services_actions_ManageAspectsService__onManageAspects(payload) {
-         if (payload && payload.item && payload.item.nodeRef)
+         if (payload && payload.node)
          {
-            // This is the array of aspects that are currently applied to the node. These will automatically
-            // be hidden from the list of available aspects to be applied...
-            var aspects = [];
-
-            // The aspects may already be available in the node data, if not they will need to be requested...
-            if (payload.item.node && payload.item.node.aspects)
-            {
-               // Aspects are already provided, use these...
-               array.forEach(payload.item.node.aspects, lang.hitch(this, this.processAspect, aspects));
-               this.showAspectsDialog(payload.item, aspects);
-            }
-            else
-            {
-               // We need to request the aspects for the current node...
-               this.onAspects(payload.item);
-            }
+            this.serviceXhr({
+               url: AlfConstants.PROXY_URI + "slingshot/doclib/aspects/node/" + payload.node.nodeRef.replace("://", "/"),
```

Comment:
Yes, very much so... if it's not, we're in serious trouble long before this point :)